lib/ansible/modules/cloud/google/gcp_forwarding_rule.py

```diff
@@ -0,0 +1,378 @@
+      U(https://cloud.google.com/compute/docs/reference/latest/forwardingRules)
+requirements:
+  - "python >= 2.6"
+  - "google-api-python-client >= 1.6.2"
+  - "google-auth >= 0.9.0"
+  - "google-auth-httplib2 >= 0.0.2"
+notes:
+  - Currently only supports global forwarding rules.
+    As such, Load Balancing Scheme is always EXTERNAL.
+author:
+  - "Tom Melendez (@supertom) <tom@supertom.com>"
+options:
+  address:
+    description:
+       - IPv4 or named IP address. Must be of the same scope (regional, global).
+         Reserved addresses can (and probably should) be used for global
+         forwarding rules. You may reserve IPs from the console or
+         via the gce_eip module.
+    required: false
+  forwarding_rule_name:
+    description:
+       - Name of the Forwarding_Rule.
+    required: true
+  port_range:
+    description:
+       - For global forwarding rules, must be set to 80 or 8080 for TargetHttpProxy, and
+         443 for TargetHttpsProxy or TargetSslProxy.
+    required: false
+  protocol:
+    description:
+       - For global forwarding rules, TCP, UDP, ESP, AH, SCTP or ICMP. Default is TCP.
+    required: false
+  region:
+    description:
+       - The region for this forwarding rule. Currently, only 'global' is supported.
+    required: false
+  target:
+    description:
+       - Target resource for forwarding rule. For global proxy, this is a Global
+         TargetProxy resource. Required for external load balancing (including Global load balancing)
+    required: false
+'''
+
+EXAMPLES = '''
+- name: Create Minimum GLOBAL Forwarding_Rule
+  gcp_forwarding_rule:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    forwarding_rule_name: my-forwarding_rule
+    protocol: TCP
+    port_range: 80
+    region: global
+    target: my-target-proxy
+    state: present
+
+- name: Create Forwarding_Rule w/reserved static address
+  gcp_forwarding_rule:
+    service_account_email: "{{ service_account_email }}"
+    credentials_file: "{{ credentials_file }}"
+    project_id: "{{ project_id }}"
+    forwarding_rule_name: my-forwarding_rule
+    protocol: TCP
+    port_range: 80
+    address: my-reserved-static-address-name
+    region: global
+    target: my-target-proxy
+    state: present
+'''
+
+RETURN = '''
+forwarding_rule_name:
+    description: Name of the Forwarding_Rule
+    returned: Always
+    type: str
+    sample: my-target-proxy
+forwarding_rule:
+    description: GCP Forwarding_Rule dictionary
+    returned: Always. Refer to GCP documentation for detailed field descriptions.
+    type: dict
+    sample: { "name": "my-forwarding_rule", "target": "..." }
+region:
+    description: Region for Forwarding Rule.
+    returned: Always
+    type: bool
+    sample: true
+state:
+    description: state of the Forwarding_Rule
+    returned: Always.
+    type: str
+    sample: present
+updated_forwarding_rule:
```

Comment:
Please remove this property since ansible already exposes `changed` so users can see whether a resource was altered during a task.